File: M1/C/pi.c
#include <stdio.h>

double compute_pi(int n){
    double res = 0;
    for (int i = 0; i < n; i++)
    {
        if (i % 2 == 0)
        {
            res = res + 1.0/(2*i + 1);
        } else
        {
            res = res - 1.0/(2*i + 1);
        }
    }
    
    return 4 * res;
}


int main(){
    int n = 10000000000;
    double pi = compute_pi(n);
    printf("%lf", pi);
}

File: M1/Rust/src/bin/pi.rs
/* use std::io; */


fn compute_pi(n:i32) -> f64{
    let mut res:f64 = 0 as f64;
    for _i in 0..n+1 {
        if _i % 2 == 0 {
            res = res + 1.0/(2.0*(_i as f64) + 1.0);
        } else {
            res = res - 1.0/(2.0*(_i as f64) + 1.0);
        }
    }
    return 4 as f64 * res;
}

fn main(){
    let n:i32 = 1000000;
    let x:f64 = compute_pi(n);
    println!("{:.64}", x);
}